packages/portfolio-contract/src/portfolio.flows.ts

```diff
@@ -699,6 +704,58 @@ export const rebalance = async (
   }
 };
 
+export const rebalanceFromTransfer = (async (
+  orch: Orchestrator,
+  ctx: PortfolioInstanceContext,
+  packet: VTransferIBCEvent['packet'],
+  kit: PortfolioKit,
+): Promise<ReturnType<LocalAccount['parseInboundTransfer']> | null> => {
+  await null;
+  const { reader } = kit;
+
+  const lca = reader.getLocalAccount();
+  const parsed = await lca.parseInboundTransfer(packet);
+  if (!parsed) {
+    return null;
+  }
+  trace('rebalanceFromTransfer parsed', parsed);
+
+  const {
+    amount,
+    extra: { receiver },
+  } = parsed;
+  const { baseAddress, query } = decodeAddressHook(receiver);
+  const { rebalance: strategy } = query;
+  if (strategy === undefined) {
+    return harden(parsed);
+  }
+
+  switch (strategy) {
+    case RebalanceStrategy.DoWhatIMean:
+    case RebalanceStrategy.PreserveExistingProportions: {
```

Comment:
we usually use literals inline and count on static analysis to keep them consistent.

Reply:
but now some parts of the contract do it one way and some do it the other

oh well. I guess I don't feel that strongly about it.